internal/webhooks/clusterclass.go

```diff
@@ -274,27 +287,66 @@ func (webhook *ClusterClass) validateRemovedMachineDeploymentClassesAreNotUsed(c
 	return allErrs
 }
 
+func (webhook *ClusterClass) validateRemovedMachinePoolClassesAreNotUsed(clusters []clusterv1.Cluster, oldClusterClass, newClusterClass *clusterv1.ClusterClass) field.ErrorList {
+	var allErrs field.ErrorList
+
+	removedClasses := webhook.removedMachineClasses(oldClusterClass, newClusterClass)
+	// If no classes have been removed return early as no further checks are needed.
+	if len(removedClasses) == 0 {
+		return nil
+	}
+	// Error if any Cluster using the ClusterClass uses a MachinePoolClass that has been removed.
+	for _, c := range clusters {
+		for _, machinePoolTopology := range c.Spec.Topology.Workers.MachinePools {
+			if removedClasses.Has(machinePoolTopology.Class) {
+				// TODO: Same as above for MachineDeployments
```

Comment:
nit: please inline the entire TODO. Just easier to find later for Killian

Reply:
Does "inline" mean to include Killian's username in the TODO like so?
`// TODO(killianmuldoon): Same as above for MachineDeployments`

Reply:
I mean the TODO comment should be just the same as above (the entire text)